otter/convergence/gathering.py

```diff
@@ -111,8 +114,44 @@ def group_id(s):
                             filter(server_predicate),
                             filter(has_group_id))
 
-    eff = get_all_server_details(changes_since)
-    return eff.on(servers_apply)
+    return get_all_server_details(changes_since).on(servers_apply)
+
+
+def all_servers(cache, changes):
+    """
+    Return all servers got by merging cache and changes since the cache was
+    last fetch
+    """
+    def sdict(servers):
+        return {s['id']: s for s in servers}
+
+    return merge(sdict(cache), sdict(changes)).itervalues()
```

Comment:
Do we have to do something more than just a dictionary merge?  Otherwise, won't the cache of deleted servers will grow indefinitely?  E.g. - if there is a deleted server in the cache that is no longer in the list of changes, should we then remove it from the next cache, which counts it as cleaned up?